adafruit_74hc595.py

```diff
@@ -97,23 +101,29 @@ def value(self, val: bool):
             self._shift_register.gpio = gpio
 
     @property
-    def direction(self):
+    def direction(self) -> "typing.Optional[digitalio.Direction]":
         """``Direction`` can only be set to ``OUTPUT``."""
         return digitalio.Direction.OUTPUT
 
     @direction.setter
-    def direction(self, val: digitalio.Direction):  # pylint: disable=no-self-use
+    def direction(  # pylint: disable=no-self-use
+        self,
+        val: digitalio.Direction,
+    ) -> None:
         """``Direction`` can only be set to ``OUTPUT``."""
         if val != digitalio.Direction.OUTPUT:
             raise RuntimeError("Digital input not supported.")
 
     @property
-    def pull(self):
+    def pull(self) -> None:
         """Pull-up/down not supported, return None for no pull-up/down."""
         return None
 
     @pull.setter
-    def pull(self, val: digitalio.Pull):  # pylint: disable=no-self-use
+    def pull(  # pylint: disable=no-self-use
+        self,
+        val: "typing.Optional[digitalio.Pull]",
```

Comment:
Since this errors if it's not `None` the typing should be that.

Reply:
Like this?

```python
    @pull.setter
    def pull(  # pylint: disable=no-self-use
        self,
        val: "typing.Optional[digitalio.Pull]",
    ) -> "typing.Union[None, typing.NoReturn]":
        """Only supports null/no pull state."""
        if val is not None:
            raise RuntimeError("Pull-up and pull-down not supported.")
```

Reply:
It should be:

```python
        val: None,
    ) -> None
```

You only need to use quotes if you're typing a [forward reference](https://peps.python.org/pep-0484/#forward-references), too!

Reply:
For reference, typing something as `NoReturn` is for when a function has NO possibility to return, like a function the loops an error message forever.  Type hints unfortunately [don't take errors raised into account](https://peps.python.org/pep-0484/#exceptions), either, so the return should just be `None`.

Reply:
Y'know.. I thought I had a reasonable handle on typing hints haha